src/stores/game.js

```diff
@@ -174,118 +174,38 @@ export const useGameStore = defineStore('game', {
-        } else {
-          this.lastEventTargetType = null;
-        }
-        if (Object.hasOwnProperty.call(newGame.lastEvent, 'chosenCard')) {
-          this.cardChosenFromScrap = newGame.lastEvent.chosenCard;
-          this.playerChoosingFromScrap = newGame.lastEvent.pNum === this.myPNum;
-        } else {
-          this.cardChosenFromScrap = null;
-          this.playerChoosingFromScrap = null;
-        }
-        if (Object.hasOwnProperty.call(newGame.lastEvent, 'discardedCards')) {
-          this.discardedCards = newGame.lastEvent.discardedCards;
-        } else {
-          this.discardedCards = null;
-        }
-      }
+      this.lastEventChange = newGame.lastEvent?.change ?? null;
+      this.lastEventOneOffRank = newGame.lastEvent?.oneOff?.rank ?? null;
+      this.lastEventTargetType = newGame.lastEvent?.oneOffTargetType ?? null;
+      this.cardChosenFromScrap = newGame.lastEvent?.chosenCard ?? null;
+      this.playerChoosingFromScrap = newGame.lastEvent?.pNum === this.myPNum ?? null;
+      this.discardedCards = newGame.lastEvent?.discardedCards ?? null;
       this.waitingForOpponentToStalemate = false;
-      if (Object.hasOwnProperty.call(newGame, 'id')) {
-        this.id = newGame.id;
-      }
-      if (Object.hasOwnProperty.call(newGame, 'turn')) {
-        this.turn = newGame.turn;
-      }
-      if (Object.hasOwnProperty.call(newGame, 'chat')) {
-        this.chat = cloneDeep(newGame.chat);
-      }
-      if (Object.hasOwnProperty.call(newGame, 'deck')) {
-        this.deck = newGame.deck?.map((card) => createGameCard(card));
-      }
-      if (Object.hasOwnProperty.call(newGame, 'scrap')) {
-        this.scrap = newGame.scrap?.map((card) => createGameCard(card));
-      }
-      if (Object.hasOwnProperty.call(newGame, 'log')) {
-        this.log = cloneDeep(newGame.log);
-      }
-      if (Object.hasOwnProperty.call(newGame, 'name')) {
-        this.name = newGame.name;
-      }
-      if (Object.hasOwnProperty.call(newGame, 'p0Ready')) {
-        this.p0Ready = newGame.p0Ready;
-      }
-      if (Object.hasOwnProperty.call(newGame, 'p1Ready')) {
-        this.p1Ready = newGame.p1Ready;
-      }
-      if (Object.hasOwnProperty.call(newGame, 'passes')) {
-        this.passes = newGame.passes;
-      }
-      if (Object.hasOwnProperty.call(newGame, 'players')) {
-        this.players = newGame.players.map((player) =>
+      this.id = newGame.id || this.id;
+      this.turn = newGame.turn || this.turn;
+      // this.chat = cloneDeep(newGame.chat);
+      this.deck = newGame.deck?.map((card) => createGameCard(card)) || this.deck;
+      this.scrap = newGame.scrap?.map((card) => createGameCard(card)) || this.scrap;
+      this.log = cloneDeep(newGame.log) || this.log;
+      this.name = newGame.name || this.name;
+      this.p0Ready = newGame.p0Ready || this.p0Ready;
+      this.p1Ready = newGame.p1Ready || this.p1Ready;
+      this.passes = newGame.passes || this.passes;
```

Comment:
Any reason to use `||` instead of `??` I don't think we want to exclude false values so I'd think we want ??

Reply:
Good point, I believe you're right it should be ??